packages/app/features/Send/InformationInput/components/TransactionFee/AbstractedTransactionFee.tsx

```diff
@@ -0,0 +1,243 @@
+import type { FC } from 'react';
+import { useEffect, useRef, useState } from 'react';
+import {
+	ActivityIndicator,
+	Image,
+	StyleSheet,
+	TouchableOpacity,
+} from 'react-native';
+import type { Token, TransactionPayload } from '@walless/core';
+import { Networks } from '@walless/core';
+import { BindDirections, modalActions, Text, View } from '@walless/gui';
+import { ChevronDown, Exclamation } from '@walless/icons';
+import type { CollectibleDocument, TokenDocument } from '@walless/store';
+import { useSnapshot } from 'valtio';
+
+import {
+	injectedElements,
+	transactionActions,
+	transactionContext,
+} from '../../../../../state/transaction';
+
+import TokenFeeDropDown from './TokenFeeDropDown';
+
+interface Props {
+	tokenForFee?: Token;
+	payload?: TransactionPayload;
+	feeText?: string;
+}
+
+export const AbstractedTransactionFee: FC<Props> = () => {
+	const [isDropped, setIsDropped] = useState(false);
+	const [isFeeLoading, setIsFeeLoading] = useState(false);
+	const [error, setError] = useState<string>('');
+
+	const { tokens, getTransactionFee, getTransactionAbstractFee, network } =
+		useSnapshot(injectedElements);
+
+	const {
+		type,
+		token,
+		nftCollection,
+		transactionFee,
+		receiver,
+		sender,
+		amount,
+		nftCollectible,
+		tokenForFee,
+	} = useSnapshot(transactionContext);
+
+	const handleSetTokenFee = (tokenForFee: Token) => {
+		transactionActions.setTokenForFee(tokenForFee as TokenDocument);
+	};
+
+	const dropdownRef = useRef(null);
+
+	let tokenForFeeName = 'Unknown';
+
+	if (network == Networks.solana) {
+		if (tokenForFee && tokenForFee.metadata?.symbol) {
+			tokenForFeeName = tokenForFee.metadata.symbol;
+		} else {
+			tokenForFeeName = 'SOL';
+		}
+	} else if (network == Networks.sui) {
+		tokenForFeeName = 'SUI';
+	} else if (network == Networks.aptos) {
+		tokenForFeeName = 'APT';
+	}
+
+	useEffect(() => {
+		if (!tokenForFee) {
+			transactionActions.setTokenForFee(tokens[0] as TokenDocument);
+		}
+
+		(async () => {
+			const payload: TransactionPayload = {
+				sender: sender,
+				receiver: receiver,
+				tokenForFee: tokenForFee as Token,
+				amount: type === 'Token' ? parseFloat(amount ?? '0') : 1,
+				token:
+					type === 'Token'
+						? (token as TokenDocument)
+						: (nftCollectible as CollectibleDocument),
+				network: token?.network as Networks,
+			};
+
+			if (payload.receiver === '' || payload.amount === 0) {
+				transactionActions.setTransactionFee(0);
+				return;
+			}
+
+			setIsFeeLoading(true);
+			let fee = await getTransactionFee(payload);
+
+			if (tokenForFee?.metadata?.symbol !== 'SOL') {
+				fee = await getTransactionAbstractFee(payload);
+			}
```

Comment:
why not check before binding `fee`? Any specific logic flow?

Reply:
i don't get what you mean

Reply:
why set `fee = getTransactionFee()` then check an independent condition and set `fee = getTransactionAbstractFee()` again?

Reply:
because if i do as you said, it's gonna be like this
``` let fee = 0;
if (...) {
fee = await getTransactionAbstractFee(payload)
} else {..}
```

it takes more lines

Reply:
why not use 
```typescript
const fee = condition ? func1() : func2();
```
beside, I find that taking more lines also better than call a func for nothing, it could cause error, consume computing power, etc.